test/functional/formatting_stack/kondo_classpath_cache.clj

```diff
@@ -0,0 +1,21 @@
+(ns functional.formatting-stack.kondo-classpath-cache
+  (:require
+   [clojure.test :refer [are deftest is testing]]
+   [formatting-stack.kondo-classpath-cache :as sut]
+   [formatting-stack.util :refer [rcomp]]))
+
+(deftest runner
+  (testing "All choices have identical arglists (and therefore can be swapped transparently)"
+    (let [arglists (->> sut/runner-mapping vals (map (rcomp find-var meta :arglists)))]
+      (assert (seq arglists))
+      (assert (every? some? arglists))
+      (is (apply = arglists))))
+
+  (let [proof (atom nil)]
+    (are [input expected] (testing "All arguments passed to the `body` are evaluated"
+                            (reset! proof [])
+                            (is (= expected
+                                   input))
+                            true)
+      @(sut/runner (swap! proof conj 1))                      [1]
+      @(sut/runner (swap! proof conj 1) (swap! proof conj 2)) [1 2])))
```

Comment:
`(do (reset! proof []) input)` complects test setup with test assertions. Great tests tend to resemble [4PT](https://thoughtbot.com/blog/four-phase-test).

Additionally, one should be able to reason about of what is evaluated first (e.g. `reset` or `expected`).

> By using = directly the reporting improves for cursive users too (since a diff can be calculated)

`are` can invoke an arbitrary amount of `is` clauses (e.g. by invoking a defn that perform multiple `is`es). Cursive should handle that and emit proper diffs (just like it should in non-`are` cases, e.g. `is` inside `doseq`)

Were you aware of that nuance? Is this a reported issue?

Reply:
> `(do (reset! proof []) input)` complects test setup with test assertions

Since `are` expands to `(is (testing "..." (reset! proof []) ...)` I'm not sure about the difference here.

> By using = directly the reporting improves for cursive users too

I was a bit quick to judge here; the reporting doesn't matter. Cursive depends on the failure containing a `=` symbol in order to show the `diff`-option. That's the case in my suggestion **and** the current code.

Reply:
> Since are expands to (is (testing "..." (reset! proof []) ...) I'm not sure about the difference here.

The difference is that under my shape `(reset! proof [])` (the setup) happens clearly _before_ `=` (the verification).

Whereas `(do (reset! proof []) input)` does not make it evident  what sexpr is evaluated first (e.g. `reset!` or `expected`).

...that is, not without understanding https://clojure.org/reference/evaluation first carefully.

> That's the case in my suggestion and the current code.

Just in case, is there something that can be improved now?

Reply:
> Whereas `(do (reset! proof []) input)` does not make it evident what sexpr is evaluated first `(e.g. reset! or expected)`.

The evaluation order is not ambiguous here. I guess it could be be unexpected for the reader that `expected` is evaluated before the reset of the proof. However: Tests should always be against a known value, which is the case here too. So the evaluation order doesn't matter.

> Just in case, is there something that can be improved now?

Nothing besides the initial suggestion.